src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (this[i] !== null && this[i] !== undefined) {
+        if (i !== this.length - 1) {
+          string += `${this[i]}${separator}`;
+        } else {
+          string += `${this[i]}`;
+        };
```

Comment:
make to if statement:
first when we add next item
second - we add separator until it's not last item